openedx/core/djangoapps/content_libraries/tests/test_views_lti.py

```diff
@@ -0,0 +1,55 @@
+"""
+Tests for LTI views.
+"""
+
+from django.test import TestCase
+from django.urls import reverse
+
+from openedx.core.djangoapps.content_libraries.constants import PROBLEM
+from openedx.core.djangoapps.content_libraries.views import LibraryBlockLtiUrlView
+
+from .base import URL_LIB_LTI_JWKS
+from .base import skip_unless_cms, ContentLibrariesRestApiTest
+
+
+
+@skip_unless_cms
+class LtiToolJwksViewTest(TestCase):
+    """
+    Test JWKS view.
+    """
+
+    def test_when_no_keys_then_return_empty(self):
+        """
+        Given no LTI tool in the database.
+        When JWKS requested.
+        Then return empty
+        """
+        response = self.client.get(URL_LIB_LTI_JWKS)
+        self.assertEqual(response.status_code, 200)
+        self.assertJSONEqual(response.content, '{"keys": []}')
+
+
+class LibraryBlockLtiUrlViewTest(ContentLibrariesRestApiTest):
+    """
+    Test generating LTI URL for a block in a library.
+    """
+
+    def test_lti_url_generation(self):
+        """
+        Test the LTI URL generated from the block ID.
+        """
+
+        library = self._create_library(
+            slug="libgg", title="A Test Library", description="Testing library", library_type=PROBLEM,
+        )
+
+        block = self._add_block_to_library(library['id'], PROBLEM, PROBLEM)
+        usage_key = str(block.usage_key)
+
+        url = reverse('content_libraries:lti-url', kwargs={"usage_key_str": usage_key})
+        expected_lti_url = f"{reverse('content_libraries:lti-launch')}?id={usage_key}"
```

Comment:
Can we have the `lti-url` and the expected launch URL hardcoded here? It would help us identify unexpected changes in these  URLs that can break backward compatibility.

Reply:
I would say as a general practice to not hardcode the launch URL. Hardcoding it won't make it harder to break backend/front-end integration, but generates additional noise on backend changes.

Reply:
Hard-coding is for capturing inadvertent changes to the URL and avoid breaking dependencies (e.g., UI). Here's one scenario I want to avoid:

1. One upon a time, a new URL is added to the public interface. Tests cover the functionality. UI consumes the URL.
1. At a later time, maybe months, or maybe years, an inadvertent change is made to the URL path (e.g., maybe an `include()` is changed in a parent URL hierarchy, maybe someone is refactoring something, etc.). Tests pass because the URL is updated in the test automatically.
1. Code is deployed. UI breaks in production or pre-production.

> Hardcoding it won't make it harder to break backend/front-end integration, but generates additional noise on backend changes.

Agreed, it won't prevent someone that is intentionally migrating APIs to break things. It prevents unexpected changes to the URL from going unnoticed. In that sense, the "noise" is actually beneficial. It creates conscious awareness you are changing a public aspect of the interface and doing it intentionally. And If something breaks due to a conscious URL change down the road, people will have an obvious clue from how public interfaces are affected.

Reply:
@jvdm first of all, I have to apologize here. I didn't realize you commented on the tests. I agree with you absolutely that tests should catch these changes and I'll commit the hard-coding here in a bit.

Reply:
Thanks for accomodating the changes!